src/goto-instrument/accelerate/polynomial.cpp

```diff
@@ -138,15 +138,11 @@ void polynomialt::from_expr(const exprt &expr)
   }
   else if(expr.id()==ID_constant)
   {
-    mp_integer mp;
-    unsigned int l;
-    constant_exprt const_expr=to_constant_expr(expr);
-
-    mp=binary2integer(const_expr.get_value().c_str(), true);
-    l=mp.to_long();
+    const constant_exprt &const_expr = to_constant_expr(expr);
+    const mp_integer mp = binary2integer(const_expr.get_value().c_str(), true);
 
```

Comment:
This should really be numeric_cast<int>(const_expr)

Reply:
Done.